mmv1/third_party/terraform/tests/resource_bigquery_data_transfer_config_test.go

```diff
@@ -10,6 +10,106 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
+func TestBigqueryDataTransferConfig_resourceBigqueryDTCParamsCustomDiffFuncForceNew(t *testing.T) {
```

Comment:
please add cases to test that the resource won't be recreated if any of these fields change but data_source_id is _not_ google_cloud_storage. (That is, make copies of the test cases but change data_source_id and have them all expect forcenew: false)

Reply:
`scheduled_query` data_source_id doesn't expect the `data_path_template` field in the `params` field. So, I've only added a test case for the `destination_table_name_template` field where data_source_id is `scheduled_query`.